### PR TITLE
fixes #6229 - validate installation media exists for new host if pxe_build

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -137,13 +137,14 @@ class Jail < ::Safemode::Jail
...
-                          :if => Proc.new { |host| host.managed && pxe_build? }
+                          :if => Proc.new { |host| host.managed && host.pxe_build? }
     validates :ip, :format => {:with => Net::Validations::IP_REGEXP}, :if => Proc.new { |host| host.require_ip_validation? }
     validates :ptable_id, :presence => {:message => N_("cant be blank unless a custom partition has been defined")},
-                          :if => Proc.new { |host| host.managed and host.disk.empty? and not Foreman.in_rake? and pxe_build? }
+                          :if => Proc.new { |host| host.managed and host.disk.empty? and not Foreman.in_rake? and host.pxe_build? }
     validates :serial, :format => {:with => /[01],\d{3,}n\d/, :message => N_("should follow this format: 0,9600n8")},
                        :allow_blank => true, :allow_nil => true
     validates :provision_method, :inclusion => {:in => PROVISION_METHODS, :message => N_('is unknown')}, :if => Proc.new {|host| host.managed?}
+    validates :medium_id, :presence => true, :if => Proc.new { |host| host.pxe_build? }
```
Comment (domcleal) [inline app/models/host/managed.rb]: host.managed too?

Comment (GregSutcliffe) [inline app/models/host/managed.rb]: :+1: this should only apply to managed hosts

Comment (isratrade) [inline app/models/host/managed.rb]: @GregSutcliffe, it's been fixed in the current commit

Comment (GregSutcliffe) [inline app/models/host/managed.rb]: Whoops, email-reading-failure :)

**Comment by GregSutcliffe** [comment]
:+1: tests fine locally, merge at will.

File: test/factories/host_related.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
     sequence(:name) { |n| "host#{n}" }
     domain
     environment
+    medium
```
Comment (domcleal) [inline test/factories/host_related.rb]: If this is optional (which I think it is for an unmanaged host), it should probably be in a trait so we keep the basic model minimal.

**Comment by isratrade** [comment]
re-committed with change to Host factory

**Comment by isratrade** [comment]
not sure of katello test failure 'Validation failed: Name Label has already been taken'
@jsherrill, can you take a look

**Comment by domcleal** [comment]
[test]

**Comment by domcleal** [comment]
Merged as 6568415 for Foreman 1.6.0, thanks @isratrade.

**Pull request closed without merging**